src/content/docs/en/guides/view-transitions.mdx

```diff
@@ -217,3 +217,34 @@ import { ViewTransitions } from 'astro:transitions';
 :::note[known limitations]
 The `morph` animation cannot be simulated in traditional CSS. So any element using this animation will not be animated.
 :::
+
+## Caveats
+
+When using the `<ViewTransitions />` component page navigations do not behave exactly the same as they do when navigating directly to a page (such as typing the URL into the address bar).
+
+### Scripts
+
+When navigating between pages with the `<ViewTransitions />` component scripts are run in sequential order to match browser behavior. If you have code that sets up global state, this state will need to take into account that the script might execute more than once. You can usually do this by checking for the global state and conditionally executing your code.
+
+```astro
+<script is:inline>
+  if(!window.SomeGlobal) {
+    window.SomeGlobal = {} // ....
+  }
+</script>
+```
+
+Module scripts are only ever executed once, because the browser keeps track of which modules are already loaded. For these scripts you do not need to worry about re-execution.
+
+#### astro:load
+
+Additionally, Astro triggers the `astro:load` event on the `document` when a new page loads. This happens both when the initial page loads and for any subsequent navigation. You can use this event to run code on every page navigation, or only once ever.
```

Comment:
```suggestion
An event listener to/that.... FILL THIS IN MATTHEW

When using the `<ViewTransitions />` component, Astro triggers the `astro:load` event on the `document` when a new page loads. This happens both when the initial page loads and for any subsequent navigation. 

Use this event to run code on every page navigation, or only once ever:
```
Since `astro:load` is new and undocumented, let's give it at least an intro explanatory sentence here. FILL THAT IN, MATTHEW!  :smile:

Also, just splitting up ideas to make the text more scannable.

Reply:
Wasn't the sentence I had introductory?

Reply:
I'll try another pass at it.